### PR TITLE
Attempt Python 3.12 support for ok-client

### PR DESCRIPTION
**Pull request by cycomachead**
This is just building off Gabe's work. I haven't done any more investigations, but tried to update CI to use recent Python

### CODE REVIEW

**Comment by cycomachead** [comment]
@LarynQi Can you or someone else approve my fork to run workflows. :)

https://github.com/cs88-website/ok-client/actions/runs/7550596522/job/20556482294
if I update CI, following Gabe's work, tests seem to pass.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: CI
 
-on: [push, pull_request]
+on: [push, pull_request, workflow_dispatch]
```
Comment (cycomachead) [inline .github/workflows/main.yml:3]: This allows manually rerunning workflows, which is handy for folks using a fork

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -102,4 +102,4 @@ jobs:
         esac
         . "${VENV_DIR}/${scripts}/activate"
         echo "Using $(python --version) on ${{ matrix.os }}" 1>&2
-        nosetests --traverse-namespace --with-coverage tests
+        nose2 -v --with-coverage tests
```
Comment (cycomachead) [inline .github/workflows/main.yml:105]: `-v` may not be necessary, but it was helpful to make sure things were running. 
I don't know what `--traverse-namespace` does, but it is not supported in nose2

File: .ok_storage.bak [hunk 3]
```diff
@@ -0,0 +1,5 @@
+"<Mock name='mock.name' id='2137274846528'>-correct", (0, 66)
```
Comment (cycomachead) [inline .ok_storage.bak:1]: Are these necessary?
Seems like not.

File: unittest.cfg [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[unittest]
+start-dir = tests
+test-file-pattern = *_test.py
```
Comment (cycomachead) [inline unittest.cfg:3]:
```suggestion
test-file-pattern = *_test.py

```

**Review by papajohn — APPROVED** [review]
Looks fine to me.

**Pull request merged**